mocks/expectations.go

```diff
@@ -0,0 +1,20 @@
+package mocks
+
+import (
+	"errors"
+)
+
+// A simple interafce that includes the testing.T methods we use to report
+// expectation violations when using the mock objects.
+type ExpectationViolationReporter interface {
```

Comment:
just `ErrorReporter`? not all errors are necessarily going to be violations of expecations